runtime/compiler/env/J9ClassEnv.cpp

```diff
@@ -642,3 +644,57 @@ J9::ClassEnv::isValueTypeClass(TR_OpaqueClassBlock *clazz)
    J9Class *j9class = reinterpret_cast<J9Class*>(clazz);
    return J9_IS_J9CLASS_VALUETYPE(j9class);
    }
+
+bool 
+J9::ClassEnv::containesZeroOrOneConcreteClass(TR::Compilation *comp, List<TR_PersistentClassInfo>* subClasses)
+   {
+   int count = 0;
+#if defined(J9VM_OPT_JITSERVER)
+   ListIterator<TR_PersistentClassInfo> j(subClasses);
+   TR_ScratchList<TR_PersistentClassInfo> subClassesNotCached(comp->trMemory());
+   
+   // Process classes caches at the server first
+   for (TR_PersistentClassInfo *ptClassInfo = j.getFirst(); ptClassInfo; ptClassInfo = j.getNext())
+      {
+      TR_OpaqueClassBlock *clazz = ptClassInfo->getClassId();
+      J9Class *j9clazz = TR::Compiler->cls.convertClassOffsetToClassPtr(clazz);
+      auto romClass = JITServerHelpers::getRemoteROMClassIfCached(TR::compInfoPT->getClientData(), j9clazz);
```

Comment:
Please take `TR::compInfoPT->getClientData()` outside the loop by storing it in a local variable. That `TR::compInfoPT` is accessing thread local storage which is expensive to get to.